homework18/src/main/java/Box.java

```diff
@@ -0,0 +1,46 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Box<T extends Fruit> {
+
+    List<T> fruits;
+
+    public Box() {
+        fruits = new ArrayList<>();
+    }
+
+    public void addFruit(T fruit) {
+        fruits.add(fruit);
+    }
+
+    public void addFruits(List<T> fruits) {
+        this.fruits.addAll(fruits);
+    }
+
+    public float getWeight() {
+        float totalWeight = 0.0F;
+        for (T fruit : fruits) {
+            totalWeight += fruit.getWeight();
+        }
+        return totalWeight;
+    }
+
+    public boolean compare(Box<?> otherBox) {
+        return Math.abs(this.getWeight() - otherBox.getWeight()) < 0.0001;
+    }
+
+    public void merge(Box<T> otherBox) {
+        if (this == otherBox) {
+            return;
+        }
+        if (this.fruits.size() == 0) {
```

Comment:
You do not need to check the size of the fruits in the current box, you should only check if the types of each fruit you're trying to put into the current box are the same.